HW04.Rmd

```diff
@@ -1,126 +1,260 @@
-```{r intent}
-
-```
-
-## Generating a boxplot visualizing the age of gun death victims, by sex. Print the average age of female gun death victims.
-
-```{r age_&_sex}
-
-```
-
-
-## How many white males with at least a high school education were killed by guns in 2012?
-
-```{r education}
-
-```
-
-
-## Which season of the year has the most gun deaths?
-
-Assume that:
-
-* Winter = January-March
-* Spring = April-June
-* Summer = July-September
-* Fall = October-December
-
-```{r season}
-
-```
-
-
-# These are more open ended questions from here on out, you could look at making multiple summaries or types of plots. 
-
-### Are whites who are killed by guns more likely to die because of suicide or homicide? How does this compare to blacks and hispanics?
-
-```{r race}
-
-```
-
-(hint maybe looking at percentages would be good)
-```{r race_2}
-
-```
-
-
-### Are police-involved gun deaths significantly different from other gun deaths? Assess the relationship between police involvement and age, police involvement and race, and the intersection of all three variables.
-
-```{r police}
-
-```
-
-Think about performing stats such as a t.test to see if differences you are plotting are statistically significant
-
-```{r stats?}
-
-
-```
-
-
-```{r police_2}
-
-```
-
-```{r stats_2?}
-
-
-```
-
-
-Finally, all together now:
-```{r police_3}
-
-```
-
-
-## Session info
-
-```{r}
-# always good to have this for reproducibility purposes
-devtools::session_info()
-```
-
+---
+title: "Exploring Gun Deaths in America"
+output: github_document
+---
+
+## Objectives
+* To explore gun death data curated by FiveThirtyEight [original data available here](https://github.com/fivethirtyeight/guns-data) by
+    * Month
+    * Intent
+    * Age and sex
+    * Education
+    * Season
+    * Intent by race
+    * Police-relatedness 
+
+## Read in the data
+
+The gun_deaths.csv is provided in the repository. 
+```{r load-data, message=FALSE, warning=FALSE}
+library(tidyverse)    # load tidyverse packages, including ggplot2
+
+# read in the data
+library(readr)
+gun_death <- read_csv('gun_deaths.csv')
+```
+
+## Generating a data frame that summarizes the number of gun deaths per month and printing using `kable()`. 
+
+Knitr is part of the knitr package. Allows you to change column names and add captions and make pretty tables in your knitted document. Hint: set the argument format = "markdown"
+
+
+```{r month, warning=FALSE}
+library(knitr)
+monthly_death <- gun_death %>%
+    count(month) %>%
+    rename(deaths = n)
+knitr::kable(monthly_death, "markdown",
+             align = 'cc',
+             caption = "Gun deaths per month")
+```
+
+
+### Generating a bar chart with human-readable labels on the x-axis. That is, each month should be labeled "Jan", "Feb", "Mar" (full or abbreviated month names are fine), not `1`, `2`, `3`.
+
+```{r month_chart}
+library(ggplot2)
+ggplot(monthly_death, aes(x = factor(month), y = deaths - 6000)) +
+    geom_bar(stat = "identity") +
```

Comment:
I am not sure if you are aware of this but another way to plot the data is to use geom_col(). If you use geom_col, then you will not need to add the additional argument of (stat = "identity"). I was initally unsure about the difference between geom_col and geom_bar when trying to answer this question but I found this short explanation really helpful! Nevertheless, both ways are able to produce the same results and that is important! 

https://ggplot2.tidyverse.org/reference/geom_bar.html#:~:text=There%20are%20two%20types%20of,data%2C%20use%20geom_col()%20instead.